featuretools/synthesis/get_valid_primitives.py

```diff
@@ -0,0 +1,77 @@
+from featuretools.primitives import AggregationPrimitive, TransformPrimitive
+from featuretools.primitives.utils import (
+    get_aggregation_primitives,
+    get_transform_primitives
+)
+from featuretools.synthesis.deep_feature_synthesis import DeepFeatureSynthesis
+from featuretools.synthesis.utils import (
+    _categorize_features,
+    get_entityset_type,
+    get_unused_primitives
+)
+
+
+def get_valid_primitives(entityset, target_entity, max_depth=2, selected_primitives=None):
+    """
+    Returns two lists of primitives (transform and aggregation) containing
+    primitives that can be applied to the specific target entity to create
+    features.  If the optional 'selected_primitives' parameter is not used,
+    all discoverable primitives will be considered.
+
+    Args:
+        entityset (EntitySet): An already initialized entityset
+        target_entity (str): Entity id of entity to create features for.
+        max_depth (int, optional): Maximum allowed depth of features.
+        selected_primitives(list[str or AggregationPrimitive/TransformPrimitive], optional):
+            list of primitives to consider when looking for valid primitives.
+            If None, all primitives will be considered
+    Returns:
+       list[AggregationPrimitive], list[TransformPrimitive]:
+           The list of valid aggregation primitives and the list of valid
+           transform primitives.
+    """
+    agg_primitives = []
+    trans_primitives = []
+    available_aggs = get_aggregation_primitives()
+    available_trans = get_transform_primitives()
+    if selected_primitives:
+        for prim in selected_primitives:
+            if not isinstance(prim, str) and not issubclass(prim, (AggregationPrimitive, TransformPrimitive)):
+                raise ValueError(f"Selected primitive {prim} is not an "
+                                 "AggergationPrimitive, TransformPrimitive, or str")
+            if not isinstance(prim, str) and issubclass(prim, AggregationPrimitive):
+                agg_primitives.append(prim)
+            elif not isinstance(prim, str) and issubclass(prim, TransformPrimitive):
+                trans_primitives.append(prim)
+            elif prim in available_aggs:
+                agg_primitives.append(available_aggs[prim])
+            elif prim in available_trans:
+                trans_primitives.append(available_trans[prim])
+            else:
+                raise ValueError(f"'{prim}' is not a recognized primitive name")
+    else:
+        entityset_type = get_entityset_type(entityset)
+        agg_primitives = [agg for agg in available_aggs.values()
+                          if entityset_type in agg.compatibility]
+        trans_primitives = [trans for trans in available_trans.values()
+                            if entityset_type in trans.compatibility]
```

Comment:
does primitive compatibility also matter for primitives in `selected_primitives` if specified? Or can we assume that if a user is including a primitive in selected primitives, it's compatible?

Reply:
If a user supplies incompatible primitives we could error, warn and ignore, or silently ignore.

Right now it errors but I could see going with one of the other approaches.  Warn seems like a nice mix of letting the user know why some primitives weren't used without stopping the process entirely